packages/plugins/src/phone/service.ts

```diff
@@ -0,0 +1,671 @@
+      payload.phone,
+      payload.refCode
+    )
+
+    if (!verification) {
+      return err({
+        code: 'INVALID_OR_EXPIRED_VERIFICATION_TOKEN' as const,
+        message: 'Invalid or expired verification token',
+      })
+    }
+
+    if (verification.value.attempt >= 5) {
+      return err({
+        code: 'REACHED_MAX_ATTEMPTS' as const,
+        message: 'Maximum OTP verification attempts reached',
+      })
+    }
+
+    const verifyStatus = await this.options.forgotPassword
+      .onOtpVerify({
+        phone: payload.phone,
+        pin: payload.pin,
+        token: verification.id,
+      })
+      .then((result) => ok(result))
+      .catch((error) => err(error))
+
+    if (verifyStatus.isErr()) {
+      return err({
+        code: 'FAILED_TO_VERIFY_OTP' as const,
+        message: 'Failed to verify OTP',
+        cause: verifyStatus.error,
+      })
+    }
+
+    if (verifyStatus.value === false) {
+      const attempt = await this.store
+        .increaseForgotPasswordVerificationAttempt(payload.phone, payload.refCode)
+        .then((result) => ok(result))
+        .catch((error) => err(error))
+
+      if (attempt.isErr()) {
+        return err({
+          code: 'FAILED_TO_INCREASE_ATTEMPT' as const,
+          message: 'Failed to increase forgot password verification attempt',
+          cause: attempt.error,
+        })
+      }
+
+      return err({
+        code: 'INVALID_OTP' as const,
+        message: 'Invalid OTP',
+        attempt: attempt.value,
+      })
+    }
+    // TODO: Should we delete the verification
+    const accountId = await this.store.getCredentialAccountIdByUserId(verification.value.userId)
+
+    if (!accountId) {
+      return err({
+        code: 'ACCOUNT_NOT_FOUND' as const,
+        message: 'Account not found',
+      })
+    }
+
+    const resetPasswordVerification = await this.store.createResetPasswordVerification({
+      value: { accountId: accountId },
+      // TODO: Customizable
+      expiredAt: new Date(Date.now() + 15 * 60 * 1000), // 15 minutes
+    })
+
+    return ok({ token: resetPasswordVerification })
+  }
+
+  async resetPassword(payload: { token: string; password: { old: string; new: string } }) {
+    const verification = await this.store.getResetPasswordVerification(payload.token)
+
+    if (!verification) {
+      return err({
+        code: 'INVALID_OR_EXPIRED_VERIFICATION_TOKEN' as const,
+        message: 'Invalid or expired verification token',
+      })
+    }
+
+    const account = await this.store.getAccountById(verification.value.accountId)
+
+    if (!account) {
+      return err({
+        code: 'ACCOUNT_NOT_FOUND' as const,
+        message: 'Account not found',
+      })
+    }
+
+    if (account.provider !== AccountProvider.CREDENTIAL) {
+      return err({
+        code: 'ACCOUNT_NOT_SUPPORTED' as const,
+        message: 'This account does not support password login',
+      })
+    }
+
+    const password = account.password
```

Comment:
Please change variable name in order to prevent the confusion
```suggestion
    const existingPassword = account.password
```